Objective-C/CBLReplicator.mm

```diff
@@ -454,6 +455,38 @@ - (void) removeChangeListenerWithToken: (id<CBLListenerToken>)token {
     }
 }
 
+- (NSSet<NSString*>*) pendingDocumentIds: (NSError**)error {
+    if (_config.replicatorType > 1) {
+        if (error)
+            *error = [NSError errorWithDomain: CBLErrorDomain
+                                         code: CBLErrorUnsupported
+                                     userInfo: @{NSLocalizedDescriptionKey: kCBLErrorMessagePullOnlyPendingDocIDs}];
+        return [NSSet set];
+    }
+    
+    if (!_repl) {
+        CBLLogInfo(Sync, @"Trying to fetch pending documentIds without a c4replicator %@", _repl);
+        return [NSSet set];
```

Comment:
I understand that we should be able to return the pending docs even when before starting the replicator and when the replicator is stopped. We probably need to change to be able to create the replicator on demand (lazy) and make sure to also free them in destructor.

Reply:
will create a separate PR with replicator change.